### PR TITLE
Resolves #1596

### LINKED ISSUE #1596: Add-ClusterResourceDependency has an incorrect example and unclear parameters
The current example for Add-ClusterResourceDependency is incorrect:
```PowerShell
Add-ClusterResourceDependency -Cluster "FileServer-(cluster1FS12)(Cluster Disk 2)" -Resource "Cluster Disk 4"
```
The generic `-Cluster` parameter indicates what cluster to connect to, not the resource to add the new dependency to.

When I checked the parameter descriptions to find the correct usage, I found that some parameters seem mutually exclusive (like `-InputObject` and `-Resource`). But this isn't called out, and the command doesn't use parameter sets to enforce that. I tried testing different parameter combinations on a 2012 R2 test cluster, and found three valid combinations. So ideally, the documentation should be updated to show those three usage options:

```PowerShell
# Example 1: Corrected version of the current example
Add-ClusterResourceDependency -Resource 'File Server (\\cluster1FS12)' -Provider 'Cluster Disk 4'

# Example 2: Pipe in the main resource, add the dependency using the -Provider parameter
Get-ClusterResource -Name 'File Server (\\cluster1FS12)' | Add-ClusterResourceDependency -Provider 'Cluster Disk 4'

# Example 3: Pipe in the main resource, add the dependency using the -Resource parameter
Get-ClusterResource -Name 'File Server (\\cluster1FS12)' | Add-ClusterResourceDependency -Resource 'Cluster Disk 4'
```
Example 2 and 3 show how the -Resource parameter can mean different things depending on how the command is run. **Based on context, `-Resource` can mean opposite things,** either
- the resource to modify (will be set to depend on the other resource), *or*
- the dependency to add (will be added to the other resource as a dependency).

I found that surprising and unintuitive, and I'd suspect that others will be confused by it as well. I haven't seen any other PowerShell commands from Microsoft with behavior like that, so it's probably worth explicitly calling out. I'm assuming that the behavior is unintentional, but is kept for backwards compatibility.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 407bae70-7d15-5c36-101d-3b17852771d7
* Version Independent ID: ee6c092a-941c-dfbe-b59d-40e2b1ea9cfe
* Content: [Add-ClusterResourceDependency (failoverclusters)](https://docs.microsoft.com/en-us/powershell/module/failoverclusters/add-clusterresourcedependency?view=win10-ps)
* Content Source: [docset/windows/failoverclusters/Add-ClusterResourceDependency.md](https://github.com/MicrosoftDocs/windows-powershell-docs/blob/live/docset/windows/failoverclusters/Add-ClusterResourceDependency.md)
* Product: **w10**
* Technology: **powershell-windows**
* GitHub Login: @andreabarr
* Microsoft Alias: **v-anbarr**

### PR DESCRIPTION
**Pull request by velkovb**
Fix example and update parameter description.
https://github.com/MicrosoftDocs/windows-powershell-docs/issues/1596

### CODE REVIEW

**Comment by e0i** [comment]
@dariomws @get-itips @tseward

FYI

@JohanFreelancer9

File: docset/windows/failoverclusters/Add-ClusterResourceDependency.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ Add-ClusterResourceDependency [[-Resource] <String>] [[-Provider] <String>] [-In
 
 ## DESCRIPTION
 The **Add-ClusterResourceDependency** cmdlet adds a resource to the list of resources on which a particular resource depends, using AND as the connector, within a failover cluster.
-Existing dependencies will remain in the list.
+Existing dependencies will remain in the list. If you specify the *InputObject* parameter the *Resource* parameter will be ignored.
```
Comment (JohanFreelancer9) [inline docset/windows/failoverclusters/Add-ClusterResourceDependency.md:35]:
```suggestion
Existing dependencies will remain on the list. If you specify the *InputObject* parameter, the *Resource* parameter will be ignored.
```

File: docset/windows/failoverclusters/Add-ClusterResourceDependency.md [hunk 2]
```diff
@@ -83,7 +83,7 @@ Accept wildcard characters: False
 ```
 
 ### -Provider
-Specifies the cluster resource on which to add a dependency.
+Specifies the cluster resource which to add as a dependency.
```
Comment (JohanFreelancer9) [inline docset/windows/failoverclusters/Add-ClusterResourceDependency.md:86]:
```suggestion
Specifies the cluster resource to add as a dependency.
```

File: docset/winserver2012-ps/failoverclusters/Add-ClusterResourceDependency.md [hunk 3]
```diff
@@ -22,8 +22,7 @@ Add-ClusterResourceDependency [[-Resource] <String>] [[-Provider] <String>] [-Cl
 ```
 
 ## DESCRIPTION
-The **Add-ClusterResourceDependency** cmdlet adds a resource to the list of resources on which a particular resource depends, using AND as the connector, within a failover cluster.
-Existing dependencies will remain in the list.
+The **Add-ClusterResourceDependency** cmdlet adds a resource to the list of resources on which a particular resource depends, using AND as the connector, within a failover cluster. Existing dependencies will remain in the list. If you specify the *InputObject* parameter the *Resource* parameter will be ignored.
```
Comment (JohanFreelancer9) [inline docset/winserver2012-ps/failoverclusters/Add-ClusterResourceDependency.md:25]:
```suggestion
The **Add-ClusterResourceDependency** cmdlet adds a resource to the list of resources on which a particular resource depends, using AND as the connector, within a failover cluster. Existing dependencies will remain on the list. If you specify the *InputObject* parameter, the *Resource* parameter will be ignored.
```

File: docset/winserver2012-ps/failoverclusters/Add-ClusterResourceDependency.md [hunk 4]
```diff
@@ -74,7 +73,7 @@ Accept wildcard characters: False
 ```
 
 ### -Provider
-Specifies the cluster resource on which to add a dependency.
+Specifies the cluster resource which to add as a dependency.
```
Comment (JohanFreelancer9) [inline docset/winserver2012-ps/failoverclusters/Add-ClusterResourceDependency.md:76]:
```suggestion
Specifies the cluster resource to add as a dependency.
```

File: docset/winserver2012r2-ps/failoverclusters/Add-ClusterResourceDependency.md [hunk 5]
```diff
@@ -30,8 +30,7 @@ Add-ClusterResourceDependency [[-Resource] <String>] [[-Provider] <String>] [-In
 ```
 
 ## DESCRIPTION
-The **Add-ClusterResourceDependency** cmdlet adds a resource to the list of resources on which a particular resource depends, using AND as the connector, within a failover cluster.
-Existing dependencies will remain in the list.
+The **Add-ClusterResourceDependency** cmdlet adds a resource to the list of resources on which a particular resource depends, using AND as the connector, within a failover cluster. Existing dependencies will remain in the list. If you specify the *InputObject* parameter the *Resource* parameter will be ignored.
```
Comment (JohanFreelancer9) [inline docset/winserver2012r2-ps/failoverclusters/Add-ClusterResourceDependency.md:33]:
```suggestion
The **Add-ClusterResourceDependency** cmdlet adds a resource to the list of resources on which a particular resource depends, using AND as the connector, within a failover cluster. Existing dependencies will remain on the list. If you specify the *InputObject* parameter, the *Resource* parameter will be ignored.
```

File: docset/winserver2012r2-ps/failoverclusters/Add-ClusterResourceDependency.md [hunk 6]
```diff
@@ -82,7 +81,7 @@ Accept wildcard characters: False
 ```
 
 ### -Provider
-Specifies the cluster resource on which to add a dependency.
+Specifies the cluster resource which to add as a dependency.
```
Comment (JohanFreelancer9) [inline docset/winserver2012r2-ps/failoverclusters/Add-ClusterResourceDependency.md:84]:
```suggestion
Specifies the cluster resource to add as a dependency.
```

**Comment by get-itips** [comment]
@velkovb 
Are you positive on this addition?

This example adds the **_cluster_** resource named "Cluster Disk 4"...

**Comment by velkovb** [comment]
> @velkovb
> Are you positive on this addition?
> 
> This example adds the **_cluster_** resource named "Cluster Disk 4"...

@get-itips This is the confusing part here. **_Resource_** is the object which dependencies will be updated, while **_Provider_** is the resource which will be added as a dependency. I tested the command and it works that way. However, if you specify **_InputObject_** for the object which dependencies will be updated, you could either use **_Resource_** or  **_Provider_** for the resource that will be added as a dependency. Furthermore, if you specify all  **_Resource_** is ignored...

**Comment by e0i** [comment]
Thank you for the productive discussion. Much appreciated.

**Comment by garycentric** [comment]
In review.

**Pull request merged**